tests/test_input_manager.py

```diff
@@ -1849,3 +1872,707 @@ def test_flush_pool(mock_input_manager: InputManager) -> None:
+
+    with patch("RUFAS.output_manager.OutputManager.add_error") as mock_om_add_error:
+        result = mock_input_manager.add_dict_variable_to_pool(
+            variable_name=variable_name,
+            data=data,
+            properties_blob_key=properties_blob_key,
+            eager_termination=False
+        )
+
+    assert result is True
+    assert mock_om_add_error.call_count == 0
+    mock_input_manager._metadata_properties_exists.assert_called_once_with(
+        variable_name=variable_name,
+        properties_blob_key=properties_blob_key)
+    mock_input_manager._add_variable_to_pool.assert_called_once_with(
+        variable_name=variable_name,
+        data=data,
+        properties_blob_key=properties_blob_key,
+        eager_termination=False,
+        variable_type='dict'
+    )
+
+    mock_input_manager.add_dict_variable_to_pool = input_manager_original_method_states["add_dict_variable_to_pool"]
+    mock_input_manager._metadata_properties_exists = input_manager_original_method_states["_metadata_properties_exists"]
+    mock_input_manager._add_variable_to_pool = input_manager_original_method_states["_add_variable_to_pool"]
+
+
+@pytest.mark.parametrize('variable_name, data, properties_blob_key', [
+    ("var1", "a", "key1"),
+    ("var2", [1, 2, 3], "key2"),
+    ("var3", 5, "key3"),
+])
+def test_add_dict_variable_to_pool_type_error(variable_name: str,
+                                              data: Dict[str, Any],
+                                              properties_blob_key: str,
+                                              mock_input_manager: InputManager,
+                                              input_manager_original_method_states: Dict[str, Callable]) -> None:
+    mock_input_manager._metadata_properties_exists = MagicMock(return_value=(True, None))
+    mock_input_manager._add_variable_to_pool = MagicMock(return_value=(True, None))
+
+    with patch("RUFAS.output_manager.OutputManager.add_error") as mock_om_add_error:
+        with pytest.raises(TypeError):
+            result = mock_input_manager.add_dict_variable_to_pool(
+                variable_name=variable_name,
+                data=data,
+                properties_blob_key=properties_blob_key,
+                eager_termination=False
+            )
+
+            assert result is None
+            assert mock_om_add_error.call_count == 1
+            mock_input_manager._metadata_properties_exists.assert_not_called()
+            mock_input_manager._add_variable_to_pool.assert_not_called()
+
+            mock_input_manager.add_dict_variable_to_pool = \
+                input_manager_original_method_states["add_dict_variable_to_pool"]
+            mock_input_manager._metadata_properties_exists = \
+                input_manager_original_method_states["_metadata_properties_exists"]
+            mock_input_manager._add_variable_to_pool = input_manager_original_method_states["_add_variable_to_pool"]
```

Comment:
```suggestion
        with pytest.raises(TypeError):
            mock_input_manager.add_dict_variable_to_pool(
                variable_name=variable_name,
                data=data,
                properties_blob_key=properties_blob_key,
                eager_termination=False
            )

        assert mock_om_add_error.call_count == 1
        mock_input_manager._metadata_properties_exists.assert_not_called()
        mock_input_manager._add_variable_to_pool.assert_not_called()

        mock_input_manager.add_dict_variable_to_pool = \
            input_manager_original_method_states["add_dict_variable_to_pool"]
        mock_input_manager._metadata_properties_exists = \
            input_manager_original_method_states["_metadata_properties_exists"]
        mock_input_manager._add_variable_to_pool = input_manager_original_method_states["_add_variable_to_pool"]
```
Nothing is returned when an error is raised